programs/stake/src/stake_state.rs

```diff
@@ -287,20 +295,31 @@ fn calculate_stake_rewards(
 
     // Drive credits_observed forward unconditionally when rewards are disabled
     // or when this is the stake's activation epoch
-    if point_value.rewards == 0
-        || (fix_activating_credits_observed && stake.delegation.activation_epoch == rewarded_epoch)
-    {
+    if point_value.rewards == 0 {
+        if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer.as_ref() {
+            inflation_point_calc_tracer(&SkippedReason::DisabledInflation.into());
+        }
+        forced_credits_update_with_skipped_reward |= true;
```

Comment:
Any (upcoming) reason this needs to be `|=` instead of just `=`? Seems more confusing than helpful in the case of a mutable bool

Reply:
well, there is no reason. changed to `=`: 6ea81896b55cb34b10f0f8c1ac3a86bc465f3092